<?xml version='1.0'?>
<sdf version='1.6'>
  <world name='default'>

    <!-- Camera Pose -->
    <gui fullscreen='0'>
      <camera name='user_camera'>
        <pose>-12 7 21 0 1.25 0</pose>
      </camera>
    </gui>

    <!-- physics -->
    <physics type='ode'>
      <max_step_size>0.001</max_step_size>
      <real_time_update_rate>1000</real_time_update_rate>
    </physics>

    <!-- sun -->
    <include>
      <uri>model://sun</uri>
    </include>

    <!-- Ground Plane -->
    <include>
      <uri>model://ground_plane</uri>
    </include>

    <!-- fei_k5__marathon -->
    <include>
      <name>fei_k5__marathon_1</name>
      <uri>model://fei_k5__cameras</uri>
    </include>

    <!-- Human 1 -->
    <include>
      <pose>-10 7 0 0 0 0</pose>
      <name>human_model_1</name>
      <uri>model://human_cylinder</uri>
    </include>
    <actor name="human_actor_1">
      <skin>
        <filename>stand.dae</filename>
      </skin>
      <script>
        <trajectory id="0" type="stand">
          <waypoint>
            <time>0</time>
            <pose>-10 7 0 0 0 0</pose>
          </waypoint>
          <waypoint>
            <time>3</time>
            <pose>-10 7 0 0 0 0</pose>
          </waypoint>
        </trajectory>
      </script>
    </actor>

    <!-- Human 2 -->
    <include>
      <pose>-6.5 9.75 0 0 0 1.5707</pose>
      <name>human_model_2</name>
      <uri>model://human_cylinder</uri>
    </include>
    <actor name="human_actor_2">
      <skin>
        <filename>stand.dae</filename>
      </skin>
      <script>
        <trajectory id="0" type="stand">
          <waypoint>
            <time>0</time>
            <pose>-6.5 9.75 0 0 0 1.5707</pose>
          </waypoint>
          <waypoint>
            <time>3</time>
            <pose>-6.5 9.75 0 0 0 1.5707</pose>
          </waypoint>
        </trajectory>
      </script>
    </actor>

    <!-- Human 3 -->
    <include>
      <pose>-6.5 12 0 0 0 -1.5707</pose>
      <name>human_model_3</name>
      <uri>model://human_cylinder</uri>
    </include>
    <actor name="human_actor_3">
      <skin>
        <filename>stand.dae</filename>
      </skin>
      <script>
        <trajectory id="0" type="stand">
          <waypoint>
            <time>0</time>
            <pose>-6.5 12 0 0 0 -1.5707</pose>
          </waypoint>
          <waypoint>
            <time>3</time>
            <pose>-6.5 12 0 0 0 -1.5707</pose>
          </waypoint>
        </trajectory>
      </script>
    </actor>

    <!-- Human 4 -->
    <include>
      <pose>-3.75 7 0 0 0 0</pose>
      <name>human_model_4</name>
      <uri>model://human_cylinder</uri>
    </include>
    <actor name="human_actor_4">
      <skin>
        <filename>stand.dae</filename>
      </skin>
      <script>
        <trajectory id="0" type="stand">
          <waypoint>
            <time>0</time>
            <pose>-3.75 7 0 0 0 0</pose>
          </waypoint>
          <waypoint>
            <time>3</time>
            <pose>-3.75 7 0 0 0 0</pose>
          </waypoint>
        </trajectory>
      </script>
    </actor>

    <!-- picture_frame -->
    <include>
      <name>picture_frame</name>
      <uri>model://picture_frame</uri>
      <pose>-1.5 7 1.2 0 -0 3.1415</pose>
    </include>


    <plugin name="link_human_1" filename="lib_link_actor_to_model.so">
      <actor>human_actor_1</actor>
      <model>human_model_1</model>
    </plugin>
    <plugin name="link_human_2" filename="lib_link_actor_to_model.so">
      <actor>human_actor_2</actor>
      <model>human_model_2</model>
    </plugin>
    <plugin name="link_human_3" filename="lib_link_actor_to_model.so">
      <actor>human_actor_3</actor>
      <model>human_model_3</model>
    </plugin>
    <plugin name="link_human_4" filename="lib_link_actor_to_model.so">
      <actor>human_actor_4</actor>
      <model>human_model_4</model>
    </plugin>

    <plugin name="people_data" filename="lib_people_publisher.so">
      <model>human_actor_1</model>
      <model>human_actor_2</model>
      <model>human_actor_3</model>
      <model>human_actor_4</model>
    </plugin>

    <plugin name="objects_data" filename="lib_objects_publisher.so">
      <model type="picture_frame">picture_frame</model>
    </plugin>

    <plugin name="locals_data" filename="lib_locals_publisher.so">
      <local name="cp1">-10 2 0 0 0 0.78</local>
      <local name="cp2">-10 11 0 0 0 -0.78</local>
      <local name="cp3">-2.75 11 0 0 0 -2.53</local>
      <local name="cp4">-2.75 2 0 0 0 2.53</local>
    </plugin>

    <plugin name="forbidden_region" filename="lib_regions.so">
      <region name="forbidden_region_object">
        <layer>3</layer>
        <color>Red</color>
        <area>-2.5 7 2 0.5</area>
      </region>
      <region name="forbidden_region_people">
        <layer>3</layer>
        <color>Red</color>
        <area>-6.5 11 0.5 2</area>
      </region>
    </plugin>

  </world>
</sdf>
